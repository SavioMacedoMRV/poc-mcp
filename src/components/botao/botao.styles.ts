import React, {ButtonHTMLAttributes} from 'react'
import styled from 'styled-components'
import {cores} from '../../resources/cores'
import {Texto} from 'components/texto'
import {margens} from 'resources/margens'
import {TextoProps} from 'components/texto/texto.styles'
import {IconeProps} from 'components/icone'

export interface BotaoProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  icone?: React.FC<React.SVGProps<SVGSVGElement>>
  texto?: string
  cor?: string
  corDesabilitado?: string
  corTexto?: string
  corTextoDesativado?: string
  textoProps?: TextoProps
  iconeProps?: Omit<IconeProps, 'icone'>
  altura?: number
  largura?: number
  espaco?: number
  aoClicar: (e?: any) => void
}

interface StyledProps extends Omit<BotaoProps, 'aoClicar' | 'corDesabilitado'> {
  $corDesabilitado?: string
}

export const BotaoStyled = styled.button<StyledProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: ${margens.xxsmall}px ${margens.small}px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: ${({espaco}) => espaco ?? margens.small}px;
  width: ${({largura}) => (largura ? `${largura}px` : 'auto')};
  height: ${({altura}) => (altura ? `${altura}px` : 'auto')};
  cursor: ${({disabled}) => (disabled ? 'default' : 'pointer')};
  opacity: ${({disabled}) => (disabled ? 0.2 : 1.0)};
  background: ${({cor}) => cor ?? cores.neutralLight};

  &:hover {
    opacity: ${({disabled}) => (disabled ? 0.2 : 0.7)};
  }
  &:active {
    opacity: ${({disabled}) => (disabled ? 0.2 : 1)};
  }

  &:focus-visible {
    outline: unset;
    border: 2px solid ${cores.neutralMedium};
  }

  &:disabled {
    background: ${({$corDesabilitado}) =>
      $corDesabilitado ?? cores.neutralLight};
  }
`

export const TextoBotao = styled(Texto).attrs({
  estilo: 'semibold',
  alturalinha: 24,
})``
