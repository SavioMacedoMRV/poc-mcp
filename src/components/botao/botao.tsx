import React from 'react'
import {BotaoProps, BotaoStyled, TextoBotao} from './botao.styles'
import {StyleSheetManager} from 'styled-components'
import {cores} from 'resources/cores'
import {Icone} from 'components/icone'

export const Botao = ({
  aoClicar,
  icone,
  texto,
  corTexto,
  corTextoDesativado,
  corDesabilitado,
  textoProps,
  iconeProps,
  espaco,
  ...props
}: Readonly<BotaoProps>) => {
  corTexto = props.disabled ? corTextoDesativado ?? cores.neutralDark : corTexto
  const corIcone = props.disabled
    ? corTextoDesativado ?? cores.neutralDark
    : iconeProps?.cor ?? corTexto
  return (
    <StyleSheetManager shouldForwardProp={prop => !!prop}>
      <BotaoStyled
        espaco={espaco}
        onClick={aoClicar}
        $corDesabilitado={corDesabilitado}
        {...props}
      >
        <TextoBotao
          data-testid={'texto-botao'}
          cor={corTexto}
          cursor={props.disabled ? 'default' : 'pointer'}
          {...textoProps}
        >
          {texto}
        </TextoBotao>
        {icone && (
          <Icone
            icone={icone}
            cor={corIcone}
            altura={iconeProps?.altura ?? 24}
            largura={iconeProps?.largura ?? 24}
            {...iconeProps}
          />
        )}
      </BotaoStyled>
    </StyleSheetManager>
  )
}
