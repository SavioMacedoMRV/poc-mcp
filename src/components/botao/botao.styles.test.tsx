import {render} from '@testing-library/react'
import * as S from './botao.styles'

describe('Botao.styles', () => {
  test('BotaoStyled DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoStyled />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot com o espaco passado', () => {
    const {container} = render(<S.BotaoStyled espaco={16} />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot com o largura passado', () => {
    const {container} = render(<S.BotaoStyled largura={16} />)
    expect(container).toMatchSnapshot()
  })
  test('BotaoStyled DEVE ser igual ao snapshot com o altura passado', () => {
    const {container} = render(<S.BotaoStyled altura={16} />)
    expect(container).toMatchSnapshot()
  })
  test('BotaoStyled DEVE ser igual ao snapshot com o disabled como true', () => {
    const {container} = render(<S.BotaoStyled disabled={true} />)
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot com o disabled como true e a $corDesabilitado com a cor passada', () => {
    const {container} = render(
      <S.BotaoStyled disabled={true} $corDesabilitado="#00ff00}" />,
    )
    expect(container).toMatchSnapshot()
  })

  test('BotaoStyled DEVE ser igual ao snapshot com o cor passada', () => {
    const {container} = render(<S.BotaoStyled cor="#ff0000" />)
    expect(container).toMatchSnapshot()
  })

  test('TextoBotao DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoBotao />)
    expect(container).toMatchSnapshot()
  })
})
