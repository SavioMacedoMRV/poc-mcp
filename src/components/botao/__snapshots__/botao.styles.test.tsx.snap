// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0"
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o altura passado 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: 16px;
  cursor: pointer;
  opacity: 1;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    altura="16"
    class="c0"
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o cor passada 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #ff0000;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0"
    cor="#ff0000"
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o disabled como true 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: default;
  opacity: 0.2;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.2;
}

.c0:active {
  opacity: 0.2;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0"
    disabled=""
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o disabled como true e a $corDesabilitado com a cor passada 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: default;
  opacity: 0.2;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.2;
}

.c0:active {
  opacity: 0.2;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #00ff00;
}

<div>
  <button
    class="c0"
    disabled=""
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o espaco passado 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0"
    espaco="16"
  />
</div>
`;

exports[`Botao.styles BotaoStyled DEVE ser igual ao snapshot com o largura passado 1`] = `
.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: 16px;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #D9DDDB;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

<div>
  <button
    class="c0"
    largura="16"
  />
</div>
`;

exports[`Botao.styles TextoBotao DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 12px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #242625;
  text-align: start;
  cursor: text;
  text-transform: none;
}

<div>
  <span
    alturalinha="24"
    class="c0 "
    estilo="semibold"
  />
</div>
`;
